src/ChronosTime.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @license       https://www.opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Chronos;
+
+use DateTimeInterface;
+use InvalidArgumentException;
+
+class ChronosTime
+{
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_SECOND = 1000000;
+
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_MINUTE = self::MICRO_PER_SECOND * 60;
+
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_HOUR = self::MICRO_PER_MINUTE * 60;
+
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_DAY = self::MICRO_PER_HOUR * 24;
+
+    /**
+     * @var int
+     */
+    protected int $microseconds;
+
+    /**
+     * @param \Cake\Chronos\Chronos|\Cake\Chronos\ChronosDate|\Cake\Chronos\ChronosTime|\DateTimeInterface|string|int $time Time string or microseconds
+     */
+    public function __construct(Chronos|ChronosDate|ChronosTime|DateTimeInterface|string|int $time = 0)
+    {
+        if (is_int($time)) {
+            assert($time >= 0);
+            $this->microseconds = $time % self::MICRO_PER_DAY;
+        } elseif ($time instanceof ChronosTime) {
+            $this->microseconds = $time->microseconds;
+        } else {
+            $this->microseconds = static::parseTime(is_string($time) ? $time : $time->format('H:i:s.u'));
+        }
+    }
+
+    /**
+     * @param string $time Time string (HH:mm(:ss(.u)))
+     * @return int
+     */
+    protected static function parseTime(string $time): int
+    {
+        if (!preg_match('/^\s*(\d{1,2})[:.](\d{1,2})(?:[:.](\d{1,2}))(?:[.](\d+))\s*$/', $time, $matches)) {
+            throw new InvalidArgumentException('Unable to parse time string.');
+        }
+
+        $microseconds = $matches[1] * self::MICRO_PER_HOUR;
+        $microseconds += $matches[2] * self::MICRO_PER_MINUTE;
+        if (isset($matches[3])) {
+            $microseconds += $matches[3] * self::MICRO_PER_SECOND;
+        }
+        if (isset($matches[4])) {
+            $microseconds += substr($matches[4], 0, 6);
+        }
+
+        return $microseconds;
+    }
+
+    /**
+     * @return int
+     */
+    public function getMicroseconds(): int
+    {
+        return $this->microseconds % self::MICRO_PER_SECOND;
```

Comment:
Should this just return the microseconds?

Reply:
This is DateTime microseconds. The same value you get with `u` format specifier.

Reply:
:man_facepalming: my bad.